src/Export/ProductService.php

```diff
@@ -275,7 +295,37 @@ protected function buildProductsWithVariantInformation(EntitySearchResult $resul
             $products->add($product);
         }
 
-        return $products;
+        if ($this->config->getMainVariant() === MainVariant::SHOPWARE_DEFAULT) {
+            return $products;
+        }
+
+        return $this->getProductsByMainVariantBasedOnConfig($products);
+    }
+
+    protected function getProductsByMainVariantBasedOnConfig(ProductCollection $products): ProductCollection
+    {
+        $mainVariantConfig = $this->config->getMainVariant();
+        $variantProducts = new ProductCollection();
+
+        foreach ($products as $product) {
+            switch ($mainVariantConfig) {
+                case MainVariant::MAIN_PARENT:
+                    $parent = $this->getParentByMainProduct($product);
+                    break;
+                case MainVariant::CHEAPEST:
+                    $parent = $this->getParentByCheapestVariant($product);
+                    break;
+                default:
```

Comment:
This should throw an exception, as this is an unknown config option. Throwing an `InvalidArgumentException` should be sufficient here.